### ISSUE TITLE
Can't download anything, it just logins, then nothing

### ISSUE THREAD

**Issue by andreapavoni** [oSumAtrIX/DownOnSpot#80]
**Describe the bug**
I've built `DownOnSpot` on my MacOS (intel) and tried to build it from source. I had to edit `Cargo.toml` by removing the `logger_env` from dependencies (and from `main.rs`) and used the official `librespot` library, according to what I've seen in issue #73 

I did the first run to generate `settings.json`, edited it with credentials and the app keys.

For context, I have a premium account.

When running the binary, this is what I get:

```
./target/release/down_on_spot <some spotify track or album URL>
Settings successfully loaded.
Continuing with spotify account: XXXXXXXXX.
Login succeeded.
```

Then nothing else.

**To Reproduce**

1. Build `DownOnSpot` binary from source
2. Run the binary giving a Spotify track or album URL
3. nothing happens

**Expected behavior**
I expected the track or album to be downloaded

- OS: MacOS 14.3.1 Sonoma

**Comment by oSumAtrIX** [comment]
I'm not sure; it's a dupe of https://github.com/oSumAtrIX/DownOnSpot/issues/53, but that should've been resolved. Maybe you are missing the shared libraries.

**Comment by andreapavoni** [comment]
@oSumAtrIX yes, you're right! It's a duplicate of the #53 followed that thread, it works like a charm!

Also, if you want, I can open a PR with those small changes I mentioned above.

I have some confidence with Rust, but almost zero experience with these libraries (I've used some spotify rust library in the past, but only for APIs), I'd be glad to help anyway.

**Comment by oSumAtrIX** [comment]
Can you please mention what exactly fixed your issue?

> Also, if you want, I can open a PR with those small changes I mentioned above.

What kind of changes are you referring to?

**Comment by oSumAtrIX** [comment]
Why would env_logger need to be removed? If there's any issue with the crate on MacOS, probably opening an issue [here](https://github.com/rust-cli/env_logger) would make more sense.

In regards to the link, parsing should indeed be fixed.

**Comment by andreapavoni** [comment]
@oSumAtrIX 

> Why would env_logger need to be removed? If there's any issue with the crate on MacOS, probably opening an issue [here](https://github.com/rust-cli/env_logger) would make more sense.

I had problems with dependencies, I've got some weird errors with the wrong versioning, but I wasn't able to figure out why. I know that package because I've used it in other projects, but I don't know why it gave errors with dependencies. I can only guess it's related to the dependency tree with other libraries used in this project.

> In regards to the link, parsing should indeed be fixed.

I can try to figure out how to parse the URL properly:

- removing the params (`?si=XXXXXXXXX`) should be easy
- removing the `/XXXX-YY/` from the URL might be more tricky, I don't know if it always follows the pattern of `<4 chars>-<2 chars>`, in that case it should be easy as well
